File: src/old/matplotlib.jl
pymatplotlib = pylazyobject(() -> pyimport("matplotlib"))
pyplot = pylazyobject(() -> pyimport("matplotlib.pyplot"))

"""
    pyplotshow([fig]; close=true, [format])

Show the matplotlib/pyplot/seaborn/etc figure `fig`, or all open figures if not given.

If `close` is true, the figure is also closed.

The `format` specifies the file format of the generated image. By default this is `pyplot.rcParams["savefig.format"]`.
"""
function pyplotshow(fig; close::Bool=true, format::String=pyplot.rcParams.get("savefig.format", "png").jl!s)
    fig = pyisinstance(fig, pyplot.Figure) ? PyObject(fig) : pyplot.figure(fig)
    io = IOBuffer()
    fig.savefig(io, format=format)
    data = take!(io)
    if format == "png"
        display(MIME"image/png"(), data)
    elseif format in ("jpg", "jpeg")
        display(MIME"image/jpeg"(), data)
    elseif format in ("tif", "tiff")
        display(MIME"image/tiff"(), data)
    elseif format == "svg"
        display(MIME"image/svg+xml"(), String(data))
    elseif format == "pdf"
        display(MIME"application/pdf"(), data)
    else
        error("Unsupported format: $(repr(format)) (try one of: png, jpg, jpeg, tif, tiff, svg, xml)")
    end
    close && pyplot.close(fig)
    nothing
end
function pyplotshow(; opts...)
    for fig in pyplot.get_fignums()
        pyplotshow(fig; opts...)
    end
end
export pyplotshow

File: juliacall/matplotlib.py
"""
Minimal matplotlib backend which shows plots using Julia's display mechanism.
"""

from matplotlib.backend_bases import Gcf, FigureManagerBase
from matplotlib.backends.backend_agg import FigureCanvasAgg
from juliacall import Base as jl

FigureCanvas = FigureCanvasAgg

def show(format=None):
    for figmanager in Gcf.get_all_fig_managers():
        figmanager.show(format=format)

FORMAT_TO_MIME = {
    'png': 'image/png',
    'jpg': 'image/jpeg',
    'jpeg': 'image/jpeg',
    'tif': 'image/tiff',
    'tiff': 'image/tiff',
    'svg': 'image/svg+xml',
    'pdf': 'application/pdf',
}

class FigureManager(FigureManagerBase):
    def show(self, format=None):
        fig = self.canvas.figure
        mime = FORMAT_TO_MIME.get(format)
        if format is None:
            jl.display(fig)
        elif mime is None:
            raise ValueError('invalid format {}, expecting one of: {}'.format(format, ', '.join(FORMAT_TO_MIME.keys())))
        else:
            jl.display(mime, fig)
